third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -15,30 +15,29 @@ use crate::{
         StructId, TypeParameter, TypeParameterKind,
     },
     symbol::{Symbol, SymbolPool},
-    ty::{PrimitiveType, Substitution, Type, TypeDisplayContext, Variance, BOOL_TYPE},
+    ty::{
```

Comment:
this doesn't seem to be part of compiler-v2 but a generic directory in tp/move. Is this shared code between old and new compiler?

Reply:
Indeed. The move-model contains the type checker for the specification language, which we are expanding to also cover the implementation language, according to the plan.

Reply:
So implementation language is properly contained in the specification language? With specification language containing arbitrary size integers etc.?

Reply:
Oh and to avoid misunderstanding: this is _not_ shared between old compiler and new compiler. It is shared between move prover, docgen etc. and new compiler.

Reply:
Oh move-prover and new-compiler. That makes a lot more sense

Reply:
I see the design direction now. We want to make prover and compiler be handled by the same code in the 90% of logic that is common to both

Reply:
So basically

parser -> AST -> type inference and verification  are shared by prover + compiler

compiler does just the code gen from the IR-AST and type annotations.
prover does the boogie or whatever generation and the formal verification

which are the different paths.

Reply:
ChatGPT, please create design doc from this discussion ;-)